### DISCUSSION TITLE
Job doesn't fail after error and can't be canceled

### DISCUSSION THREAD

**Discussion by gscho** [concourse/concourse#5889] [Help & Support]
## Summary

I'm building Habitat packages on my concourse build system. 

If the job fails in error, the concourse job hangs. It cannot be canceled either. 

I've tried canceling it by clicking the red X in the UI and also via `fly abort-build`.

```
❯ fly -t ci abort-build -b 18141
build successfully aborted
```

Additionally there is a `timeout: 20m` set in the task.

My build task has this at the top `set -eo pipefail`.

## Steps to reproduce

I'll try to come up with a sanitized set of reproducible steps if that's required.

## Expected results

If the job is hanging for any reason I would expect I can abort the build via one of the above mentioned methods.

## Actual results

Concourse job remains in a started state forever and cannot be aborted.

## Additional context

The build container I'm using looks like this

Dockerfile:

```
FROM habitat/default-studio-x86_64-linux:1.6.56

RUN hab license accept && \
hab pkg install core/coreutils -b && hab pkg install core/util-linux -b

RUN mkdir /tasks
ADD tasks /tasks

RUN chmod -R 755 /tasks
```

## Triaging info

* Concourse version: 5.2.0
* Browser (if applicable): Firefox

**Comment by vito** [comment]
Thanks for the report! Steps to reproduce are indeed required though; converting this to a Discussion until that can be provided.

FWIW your Concourse version (v5.2.0) is quite old - does this problem still occur on the latest version? I think we've fixed a few things about aborting since then.

Side note: aborting is asynchronous - clicking the red 'X' and running `fly abort-build` will mark it as aborted, but Concourse still needs to dot its i's and cross its t's to make sure the workloads actually stop, and that can take some time if e.g. workers are misbehaving. I would expect this to resolve in ~15 minutes (worst case), but if it's *still* stuck, I would recommend taking a [stack dump](https://github.com/concourse/concourse/wiki/Taking-a-Stack-Dump).
